### PR TITLE
06650 - MigrationTestingTool compatible with v0.38 saved state

### LINKED ISSUE #6650: Update MigrationTestingTool JRS test with state saved from v0.38
PR #6634 drops support for an old version of SigSet.   
* The saved state used int the MigrationTestingTool needs to be updated to not use the older serialization scheme for SigSet.   
* The MigrationTestingTool app needs to be updated to be compatible with the saved state.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
Along with uploading the saved state to the appropriate test platform, this code changes updates MigrationTestingTool to be compatible with v0.38 and later saved states. 

Fixes #6650

### CODE REVIEW

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java [hunk 1]
```diff
@@ -245,7 +247,7 @@ public void init(
         }
 
         // FUTURE WORK: this needs to be updated once we create the next saved state for the MigrationTestingTool
-        if (previousSoftwareVersion != SoftwareVersion.NO_VERSION) {
+        if (!Objects.equals(previousSoftwareVersion, SOFTWARE_VERSION)) {
             logger.error(
                     MARKER,
                     "previousSoftwareVersion was {} when expecting SoftwareVersion.NO_VERSION",
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java:253]: this message should be updated, as `NO_VERSION` is not what's being expected

**Review by alittley — APPROVED** [review]

**Review by cody-littley — COMMENTED** [review]
LGTM once the string in the error message is fixed

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java:253]: whoops!  will be fixed.

**Review by lpetrovic05 — DISMISSED** [review]

**Review by alittley — DISMISSED** [review]

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java [hunk 2]
```diff
@@ -245,11 +247,12 @@ public void init(
         }
 
         // FUTURE WORK: this needs to be updated once we create the next saved state for the MigrationTestingTool
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java:249]: can this `FUTURE WORK` note be considered complete?

**Review by alittley — APPROVED** [review]

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java:249]: I think so, yes.

Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java]: removed

**Review by alittley — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**